core/src/client/mod.rs

```diff
@@ -734,4 +740,72 @@ impl StarknetClient for StarknetClientImpl {
 
         Ok(Address::from(evm_address_sliced))
     }
+
+    /// Returns token balances for a specific address given a list of contracts.
+    ///
+    /// # Arguments
+    ///
+    /// * `address(Address)` - specific address
+    /// * `contract_addresses(Vec<Address>)` - List of contract addresses
+    ///
+    /// # Returns
+    ///
+    ///  * `token_balances(TokenBalances)` - Token balances
+    ///
+    /// `Ok(<TokenBalances>)` if the operation was successful.
+    /// `Err(KakarotClientError)` if the operation failed.
+    async fn get_token_balances(
+        &self,
+        address: Address,
+        contract_addresses: Vec<Address>,
+    ) -> Result<TokenBalances, KakarotClientError> {
+        let mut token_balances = Vec::new();
+        let starknet_user_address = self
```

Comment:
I wonder if we should do the conversion here,

essentially a user sends a `balanceOf()` read to Kakarot itself, meaning that Kakarot should be able to perform the conversion from EVM to Starknet address itself

Reply:
Because if at some point in Solidity someone writes:

```solidity

let myBalance = MyErc20.balanceOf(address(this))

```

address(this) is a 160-bit EVM address, but Kakarot will convert it inside its execution context, so its execute_at_address' responsibility to perform this conversion

Reply:
Ah yes ok